### PR TITLE
[Doc] Update the Branches doc with LLVM Project info

### PR DESCRIPTION
**Pull request by shahmishal**

### CODE REVIEW

File: docs/Branches.md [hunk 1]
```diff
@@ -2,51 +2,45 @@
 
 ## The Development Branches
 
-| Swift  | LLVM*  | LLDB
-| ------ | ------ | ------
-| master | stable | stable
+| Swift  | LLVM Project |
+| ------ | ------       |
```
Comment (jrose-apple) [inline docs/Branches.md]: Nitpick: the dashes should extend under the whole title.

File: docs/Branches.md [hunk 2]
```diff
@@ -56,15 +50,13 @@ LLDB's `upstream-with-swift` has *both* aspects: changes are automatically merge
...
 ## Committing
 
 - Swift: new commits go to `master`
 
-- LLVM/Clang/compiler-rt: new commits go to `upstream-with-swift`
-
-- LLDB: new commits go to `stable`
+- LLVM Project: new commits go to `swift/master-next`
 
 ...then cherry-pick to the release branch (`swift-x.y-branch`) if necessary, following the appropriate release process. (Usually this means filling out a standard template, finding someone to review your code if that hasn't already happened, and getting approval from that repo's *release manager.)*
```
Comment (jrose-apple) [inline docs/Branches.md]: This should probably say "`swift-x.y-branch` or `swift/swift-x.y-branch`".

File: docs/Branches.md [hunk 3]
```diff
@@ -75,13 +67,7 @@ Some branches are *automerged* into other branches, to keep them in sync. This i
...
-- `swift-x.y-branch` (the *latest* release branch) is automerged into `stable`
-- llvm.org's `master` is automerged into `upstream-with-swift`
-- llvm.org's release branch *may* be automerged into `swift-x.y-branch`, if they are in sync
-
-### LLDB
-- `swift-x.y-branch` (the *latest* release branch) is automerged into `stable`
-- `stable` is automerged into `upstream-with-swift`
-- llvm.org's `master` is *also* automerged into `upstream-with-swift`
+### LLVM Project
+- `swift-x.y-branch` (the *latest* release branch) is automerged into `swift/master`
```
Comment (jrose-apple) [inline docs/Branches.md]: This should say `swift/swift-x.y.branch` (here and below).

**Comment by shahmishal** [comment]
@swift-ci smoke test

File: docs/Branches.md [hunk 4]
```diff
@@ -68,6 +68,6 @@ Some branches are *automerged* into other branches, to keep them in sync. This i
 - `master` is automerged into `master-next`
 
 ### LLVM Project
-- `swift-x.y-branch` (the *latest* release branch) is automerged into `swift/master`
+- `swift/swift-x.y-branch` (the *latest* release branch) is automerged into `swift/master`
 - llvm.org's `master` is automerged into `swift/master-next`
 - llvm.org's release branch *may* be automerged into `swift-x.y-branch`, if they are in sync
```
Comment (jrose-apple) [inline docs/Branches.md]: Here too, sorry.

Comment (shahmishal) [reply docs/Branches.md]: Fixed, thanks!

**Pull request merged**